 SeeingSharp.Multimedia/Core/RenderLoop.cs | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/SeeingSharp.Multimedia/Core/RenderLoop.cs b/SeeingSharp.Multimedia/Core/RenderLoop.cs
index d1fef7c..d41cca4 100644
--- a/SeeingSharp.Multimedia/Core/RenderLoop.cs
+++ b/SeeingSharp.Multimedia/Core/RenderLoop.cs
@@ -1099,6 +1099,9 @@ namespace SeeingSharp.Multimedia.Core
 
                 // Update flag indicating that last render was successful
                 m_lastRenderSuccessfully = true;
+
+                // Fire Rendered event.
+                this.OnRendered(EventArgs.Empty);
             }
             finally
             {
@@ -1106,6 +1109,14 @@ namespace SeeingSharp.Multimedia.Core
             }
         }
 
+        public event EventHandler Rendered;
+
+        protected virtual void OnRendered(EventArgs e) {
+            if (this.Rendered!=null) {
+                this.Rendered(this,e);
+            }
+        }
+
         /// <summary>
         /// Deregisters this view from the render loop.
         /// </summary>
