 SeeingSharp.Multimedia/Views/SeeingSharpRendererControl.cs | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/SeeingSharp.Multimedia/Views/SeeingSharpRendererControl.cs b/SeeingSharp.Multimedia/Views/SeeingSharpRendererControl.cs
index 1705562..f5a6e8a 100644
--- a/SeeingSharp.Multimedia/Views/SeeingSharpRendererControl.cs
+++ b/SeeingSharp.Multimedia/Views/SeeingSharpRendererControl.cs
@@ -452,7 +452,12 @@ namespace SeeingSharp.Multimedia.Views
 
             // Check parent form
             Form parentForm = this.GetParentForm();
-            if(parentForm != null)
+            if(parentForm == null)
+            {
+                // Control's parent chain is broken -> it is not visible.
+                return false;
+            }
+            else
             {
                 if(parentForm.WindowState == FormWindowState.Minimized) { return false; }
             }
